source/construct-languages.rq

```diff
@@ -37,17 +34,22 @@ construct {
                 bind(replace(replace(?altLabels, ?skipN, ""), " *\\|.*$", "") as ?altLabel)
             }
             optional { ?s rdfs:comment ?comment }
-            optional { ?s kbv:ISO639-2-T ?termLangCode . }
-            optional { ?s kbv:ISO639-3 ?iso_639_3 . }
         } union {
-            {?s skos:notation ?locCode .} union {?s kbv:ISO639-2-T ?locCode .}  
-            bind(iri(concat('http://id.loc.gov/vocabulary/iso639-2/', ?locCode)) as ?locLang)
+            ?s skos:notation ?code .
+            filter(datatype(?code) = xsd:string)
+            bind(strdt(?code, kbv:ISO639-2) as ?langCode)
+        } union {
+            ?s skos:notation ?locCode .
+            filter(
+                datatype(?locCode) = kbv:ISO639-2-T ||
+                datatype(?locCode) = xsd:string
+            )
+            bind(iri(concat('http://id.loc.gov/vocabulary/iso639-2/', str(?locCode))) as ?locLang)
             graph <http://id.loc.gov/vocabulary/languages> {
-                optional {
+                {
                     filter(!contains(?prefLabel, "|"))
                     ?locLang skos:prefLabel ?prefLabel
-                }
-                optional {
+                } union {
```

Comment:
Using `union` appears more performant than `optional` when the contents is complementary in nature. (Which seems logical for join mechanics.)